core/chains/nano.go

```diff
@@ -2,43 +2,146 @@ package chains
 
 import (
 	"encoding/json"
+	"fmt"
 	"log"
+	"math/big"
 	"net/http"
+	"sort"
+	"strconv"
+
+	"github.com/xenowits/nakamoto-coefficient-calculator/core/utils"
 )
 
-// NanoStats represents the structure for Nano's stats data from nanocharts.
-type NanoStats struct {
-	Stats struct {
-		C1s struct {
-			N int `json:"n"`
-		} `json:"c1s"`
-	} `json:"stats"`
+type NanExplorerResponse struct {
+	Rep []struct {
+		Account string `json:"account"`
+		Weight  string `json:"weight"`
+	} `json:"rep"`
+	OnlineStakeTotal string `json:"online_stake_total"`
 }
 
-func Nano() (int, error) {
-	chartsURL := "https://nanocharts.info/data/nanocharts.json"
+type EntityResponse struct {
+	Timestamp int64 `json:"timestamp"`
+	Entities  []struct {
+		Entity          string   `json:"entity"`
+		Representatives []string `json:"representatives"`
+	} `json:"entities"`
+}
 
-	log.Println("Fetching data for Nano")
+const (
```

Comment:
@BrazyDevelopment Please add a detailed comment based on the discussion so far on why the threshold should be 67%.